rdr_service/model/participant_summary.py

```diff
@@ -56,6 +69,94 @@
+             questionnaire_on_lifestyle_authored,
+             COALESCE(consent_for_electronic_health_records_authored, consent_for_study_enrollment_authored),
+             COALESCE(consent_for_dv_electronic_health_records_sharing_authored, consent_for_study_enrollment_authored)
+            ),
+            LEAST(COALESCE(sample_status_1ed10_time, '9999-01-01'),
+                COALESCE(sample_status_2ed10_time, '9999-01-01'),
+                COALESCE(sample_status_1ed04_time, '9999-01-01'),
+                COALESCE(sample_status_1sal_time, '9999-01-01'),
+                COALESCE(sample_status_1sal2_time, '9999-01-01')
+            )
+        )
+        ELSE NULL
+     END
+"""
+
+# SQL Conditional for participant's retention eligibility computed column (1 = NOT_ELIGIBLE, 2 = ELIGIBLE)
+# TO DO:  This should be updated to verify no deceased status when that is implemented (see: DA-1601)
+_COMPUTE_RETENTION_ELIGIBLE_SQL = """
+    CASE WHEN
+      consent_for_study_enrollment = 1
+      AND (consent_for_electronic_health_records = 1 OR consent_for_dv_electronic_health_records_sharing = 1)
+      AND questionnaire_on_the_basics = 1
+      AND questionnaire_on_overall_health = 1
+      AND questionnaire_on_lifestyle = 1
+      AND samples_to_isolate_dna = 1
+      AND withdrawal_status = 1
+      AND suspension_status = 1
+      AND deceased_status = 0
+    THEN 2 ELSE 1
+    END
+"""
+
+# SQL for calculating the date when a participant gained retention eligibility
+# Null unless the participant meets the retention-eligible requirements (above) and a qualifying test sample time
+# is present.  Otherwise, find the last of the consent / module authored dates and the earliest of the qualifying
+# DNA test samples.  The retention eligibility date is the later of those two
+_COMPUTE_RETENTION_ELIGIBLE_TIME_SQL = """
+     CASE WHEN retention_eligible_status = 2 AND
+          COALESCE(sample_status_1ed10_time, sample_status_2ed10_time, sample_status_1ed04_time,
+                 sample_status_1sal_time, sample_status_1sal2_time, 0) != 0
+        THEN GREATEST(
+            GREATEST (consent_for_study_enrollment_authored,
+             questionnaire_on_the_basics_authored,
+             questionnaire_on_overall_health_authored,
+             questionnaire_on_lifestyle_authored,
+             COALESCE(consent_for_electronic_health_records_authored, consent_for_study_enrollment_authored),
+             COALESCE(consent_for_dv_electronic_health_records_sharing_authored, consent_for_study_enrollment_authored)
+            ),
+            LEAST(COALESCE(sample_status_1ed10_time, '9999-01-01'),
+                COALESCE(sample_status_2ed10_time, '9999-01-01'),
+                COALESCE(sample_status_1ed04_time, '9999-01-01'),
+                COALESCE(sample_status_1sal_time, '9999-01-01'),
+                COALESCE(sample_status_1sal2_time, '9999-01-01')
+            )
+        )
+        ELSE NULL
+     END
+"""
```

Comment:
The definition of `_COMPUTE_RETENTION_ELIGIBLE_SQL` is duplicated on lines 74 and 118. Same for `_COMPUTE_RETENTION_ELIGIBLE_TIME_SQL` on lines 93 and 137.

Reply:
Ah, good catch, that may caused by the conflict merge, I will remove it.

Reply:
Done